src/content/app/tools/blast/components/blast-submission-header/BlastSubmissionHeader.tsx

```diff
@@ -81,48 +93,71 @@ export const BlastSubmissionHeader = (props: Props) => {
 
   const handleDeletion = () => {
     dispatch(deleteBlastSubmission(submissionId));
+
+    //if nothing more is left on the page after the submission is deleted, go to the new BLAST form (This might change depending on UX)
+    if (
+      blastView === 'submission-results' ||
+      (blastView === 'unviewed-submissions' &&
+        unviewedBlastSubmissions.length === 1) ||
+      (blastView === 'submissions-list' && viewedBlastSubmissions.length === 1)
+    ) {
+      navigate(urlFor.blastForm());
+    }
   };
 
   return (
-    <div className={styles.grid}>
-      <div>{blastProgram}</div>
-      <div className={styles.submissionName}>
-        {/* placeholder for submission name */}
-      </div>
-      <div className={styles.submissionDetails}>
-        <span className={styles.submissionIdLabel}>Submission</span>
-        <span>{submissionId}</span>
-        <span className={styles.editSubmission} onClick={editSubmission}>
-          Edit/rerun
-        </span>
-        <span className={styles.timeStamp}>
-          <span>{submissionTime}</span>
-          <span className={styles.timeZone}>GMT</span>
-        </span>
-        {sequenceCount && sequenceCount > 1 && (
-          <ShowHide
-            className={styles.showHide}
-            isExpanded={props.isExpanded || false}
-            onClick={() =>
-              props.toggleExpanded && props.toggleExpanded(!props.isExpanded)
-            }
+    <>
+      <div className={styles.grid}>
+        <div>{blastProgram}</div>
+        <div className={styles.submissionName}>
+          {/* placeholder for submission name */}
+        </div>
+        <div className={styles.submissionDetails}>
+          <span className={styles.submissionIdLabel}>Submission</span>
+          <span>{submissionId}</span>
+          <span className={styles.editSubmission} onClick={editSubmission}>
+            Edit/rerun
+          </span>
+          <span className={styles.timeStamp}>
+            <span>{submissionTime}</span>
+            <span className={styles.timeZone}>GMT</span>
+          </span>
+          {sequenceCount && sequenceCount > 1 && (
+            <ShowHide
+              className={styles.showHide}
+              isExpanded={props.isExpanded || false}
+              onClick={() =>
+                props.toggleExpanded && props.toggleExpanded(!props.isExpanded)
```

Comment:
This also can be written as `props.toggleExpanded?.(!props.isExpanded)`

Reply:
Thanks for reviewing @imransl. But this is not something that i have changed in this PR (it came up in the diff since i added some html to the same container and shifted the tabbing of the code). 

I can make that change but not sure whats the impact and are we sure this is correct (the `.` after the `props.toggleExpanded` doesn't feel right, its not an object here but a function

Reply:
>  are we sure this is correct (the . after the props.toggleExpanded doesn't feel right

This is called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#syntax), a recent (two years or thereabouts) addition to the javascript syntax.

Reply:
I made the suggested change @imransl. could you please review and approve if its all good. Thanks